src/filtering/markov_kernel.jl

```diff
@@ -87,18 +87,38 @@ function marginalize_cov!(
     C_DxD::AbstractMatrix,
     C_3DxD::AbstractMatrix,
 )
-    D = length(x_curr.μ)
+    _D = size(x_curr.Σ, 1)
```

Comment:
This hints at a problem: `x_curr` is not really well defined anymore as the covariance might really only be the right factor here. I still think the overall code is quite reasonable as it handles both cases (full cov and only right factor) quite well, but idk. Maybe think about what should happen for block-diagonals, and then decide how to proceed.

Reply:
Since this operates only on covariances, maybe just pass covariances in the function signature?